libs/designsystem/menu/src/menu.component.ts

```diff
@@ -75,33 +83,261 @@ export class MenuComponent implements AfterViewInit, OnDestroy {
   public buttonContainerElement: ElementRef<HTMLElement> | undefined;
 
   @ViewChild('defaultButton', { read: ElementRef })
-  public defaultButtonElement: ElementRef<HTMLElement> | undefined;
+  public defaultButtonElement: ElementRef<HTMLButtonElement> | undefined;
 
   @ContentChild(ButtonComponent, { read: ElementRef }) public userProvidedButton:
-    | ElementRef<HTMLElement>
+    | ElementRef<HTMLButtonElement>
     | undefined;
 
   @ViewChild(FloatingDirective)
-  private floatingDirective: FloatingDirective;
+  private floatingMenu: FloatingDirective;
 
-  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
+  @ContentChildren(ItemComponent, { read: ElementRef }) public kirbyItems: QueryList<
+    ElementRef<HTMLElement>
+  >;
+
+  @ContentChildren(ItemComponent) public kirbyItemComponents: QueryList<ItemComponent>;
 
+  public floatingMenuIsShown: boolean = false;
+  public FloatingOffset: typeof FloatingOffset = FloatingOffset;
   private scrollListenerDisposeFn: EventListenerDisposeFn;
+  private focusedIndex = -1;
+
+  @HostListener('keydown', ['$event'])
+  _onKeydown(event: KeyboardEvent) {
+    if (this.kirbyItems.length === 0) {
+      console.warn('[Kirby] No items found within menu');
+      return;
+    }
+    if (this.floatingMenuIsShown) {
+      this.handleKeyDownForOpenedMenu(event);
+    } else {
+      this.handleKeyDownForClosedMenu(event);
+    }
+  }
+
+  private preventDefaultAndStopImmediatePropagation(event: KeyboardEvent) {
+    event.stopImmediatePropagation();
+    event.preventDefault();
+  }
+
+  private getFirstInteractiveElement(el: HTMLIonItemElement) {
+    return el.querySelector<HTMLIonToggleElement | HTMLIonRadioElement | HTMLIonCheckboxElement>(
+      'ion-toggle:not([disabled]), ion-checkbox:not([disabled]), ion-radio:not([disabled])'
+    );
+  }
+
+  private handleKeyDownForClosedMenu(event: KeyboardEvent) {
+    const key = event.key;
+    switch (key) {
+      case ' ':
+      case 'Enter':
+      case 'ArrowDown':
+        this.focusedIndex = 0;
+        this.floatingMenu.show();
+        this.focusItem();
+        this.preventDefaultAndStopImmediatePropagation(event);
+        break;
+      case 'ArrowUp':
+        this.focusedIndex = this.kirbyItems.length - 1;
+        this.floatingMenu.show();
+        this.focusItem();
+        this.preventDefaultAndStopImmediatePropagation(event);
```

Comment:
To me it reads better, if the order is "stop the default -> do this instead..."
Don't know if you agree? 🤔 
```suggestion
        this.preventDefaultAndStopImmediatePropagation(event);
        this.focusedIndex = 0;
        this.floatingMenu.show();
        this.focusItem();
        break;
      case 'ArrowUp':
        this.preventDefaultAndStopImmediatePropagation(event);
        this.focusedIndex = this.kirbyItems.length - 1;
        this.floatingMenu.show();
        this.focusItem();
```